org.eclipse.scanning.event/src/org/eclipse/scanning/event/QueueReader.java

```diff
@@ -79,9 +79,9 @@ public QueueReader(IEventConnectorService service, Comparator<? super T> compara
 	 * @throws Exception
 	 */
 	public List<T> getBeans(final URI uri, final String queueName, final Class<T> beanClass) throws Exception {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 19 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)